### PR TITLE
Development hes

### PR DESCRIPTION
**Pull request by nicolas-heigvd**
⚠️ MUST be used with the [new base image](https://github.com/yverdon/docker-geocity/pull/6).

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # This is an example configuration for the stack.
 # Copy this to `.env` and adapt to your needs.
 
+BASE_URL=localhost
+ENABLE_SSL=False
```
Comment (monodo) [inline .env.example:5]: SSL must be enabled by default

File: geocity/settings.py [hunk 2]
```diff
@@ -26,11 +26,25 @@
...
+    SESSION_COOKIE_SECURE = True
+    CSRF_COOKIE_SECURE = True
+
+# https://stackoverflow.com/a/71194288/6630397
+if ENABLE_SSL:
+    CSRF_TRUSTED_ORIGINS = [
+        f"https://*.{os.getenv('BASE_URL')}",
+        "https://*.127.0.0.1",
+    ]
+
```
Comment (monodo) [inline geocity/settings.py:47]: SSL MUST be enabled by default, thus, invert logic

File: requirements_dev.txt [hunk 3]
```diff
@@ -12,10 +12,6 @@ asttokens==2.0.5
     # via stack-data
 backcall==0.2.0
     # via ipython
-backports-zoneinfo==0.2.1
-    # via
-    #   -r requirements.txt
-    #   django
```
Comment (monodo) [inline requirements_dev.txt:18]: Did pip-compile remove these lines ?

File: requirements.txt [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.10
 # by the following command:
 #
 #    pip-compile requirements.in
 #
 asgiref==3.5.2
     # via django
-backports-zoneinfo==0.2.1
-    # via django
```
Comment (monodo) [inline requirements.txt:10]: Did pip-compile remove these lines ?

File: Dockerfile [hunk 5]
```diff
@@ -1,13 +1,13 @@
-FROM sitdocker/geocity-base:v1.3.2.2021
+#FROM sitdocker/geocity-base:v1.3.2.2021
+FROM geocity-base:latest
```
Comment (monodo) [inline Dockerfile:2]: Please match a release. This introduces the risk of pulling an image that is not aligned with the python libs and crash the whole application.

**Review by monodo — CHANGES_REQUESTED** [review]

File: .env.example [hunk 6]
```diff
@@ -1,6 +1,8 @@
 # This is an example configuration for the stack.
 # Copy this to `.env` and adapt to your needs.
 
+BASE_URL=localhost
```
Comment (monodo) [inline .env.example:4]: I would not do that bug fix this issue instead: https://github.com/yverdon/geocity/issues/525

**Review by monodo — CHANGES_REQUESTED** [review]

File: requirements_dev.txt (hunk 3 above)
Comment (nicolas-heigvd) [reply requirements_dev.txt:18]: This file was not changed manually.

Instead, it was generated by pip-compile `6.12.2` with Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] on linux:
```
pip-compile requirements_dev.in
```

File: requirements.txt (hunk 4 above)
Comment (nicolas-heigvd) [reply requirements.txt:10]: This file was not changed manually.

Instead, it was generated by pip-compile `6.12.2` with Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] on linux:
```
pip-compile requirements.in
```

File: Dockerfile (hunk 5 above)
Comment (nicolas-heigvd) [reply Dockerfile]: Indeed. It has been fixed by commit: https://github.com/nicolas-heigvd/geocity/commit/30d985ac2e9f0cacf428c5ad833b9c1d5de9d374

File: .env.example (hunk 6 above)
Comment (nicolas-heigvd) [reply .env.example]: Fixed by commit: https://github.com/nicolas-heigvd/geocity/commit/ed9dd0929e9dffc58bfb330a4522c300182d4174

File: geocity/settings.py (hunk 2 above)
Comment (nicolas-heigvd) [reply geocity/settings.py]: Fixed by commits:
- https://github.com/nicolas-heigvd/geocity/commit/e52b53831ac54f20aef2094ea4e804b461d2a238
- https://github.com/nicolas-heigvd/geocity/commit/40698e3b0d28ad7707c4e00ffd9259b87bad7158

File: .env.example (hunk 1 above)
Comment (nicolas-heigvd) [reply .env.example]: Fixed by commit: https://github.com/nicolas-heigvd/geocity/commit/e52b53831ac54f20aef2094ea4e804b461d2a238

**Comment by nicolas-heigvd** [comment]
@monodo all requests have been fulfilled.
[`tzdata` ](https://pypi.org/project/tzdata/)has also been added as a new Python dependency.

**Comment by monodo** [comment]
Could you merge with latest develop banch ? Thanks!

**Comment by nicolas-heigvd** [comment]
@rbovard 

**won't fix** changes related to the compose v1 CLI because it's **deprecated**:


> From the end of June 2023 Compose V1 won’t be supported anymore and will be removed from all Docker Desktop versions.
> 
> Make sure you switch to [Compose V2](https://docs.docker.com/compose/compose-file/) with the docker compose CLI plugin or by activating the Use Docker Compose V2 setting in Docker Desktop. For more information, see the [Evolution of Compose](https://docs.docker.com/compose/compose-v2/)
> 
> The new Compose V2, which supports the compose command as part of the Docker CLI, is now available.
> 
> Compose V2 integrates compose functions into the Docker platform, continuing to support most of the previous docker-compose features and flags. You can run Compose V2 by replacing the hyphen (-) with a space, using docker compose, instead of docker-compose.
> 

Further readings: 
- https://github.com/docker/compose-cli/issues/1404#issuecomment-792374011 
- https://docs.docker.com/compose/reference/
- https://docs.docker.com/compose/compose-v2/
- https://docs.docker.com/compose/#compose-v2-and-the-new-docker-compose-command

**Comment by nicolas-heigvd** [comment]
> Could you merge with latest develop banch ? Thanks!

@monodo done: https://github.com/yverdon/geocity/pull/749

**Pull request closed without merging**